code/modules/awaymissions/mission_code/ruins/watcher_grave.dm

```diff
@@ -0,0 +1,256 @@
+/obj/effect/mob_spawn/corpse/goliath/pierced
+	corpse_description = "Seems to have been pierced through the heart by a Watcher spike."
+	naive_corpse_description = "It's got a pretty big boo-boo, might need one of the large plasters."
+
+/obj/effect/mob_spawn/corpse/watcher/goliath_chewed
+	corpse_description = "Prior to its death, it was badly mangled by the jaws of a Goliath."
+	naive_corpse_description = "It's all tuckered out after playing rough with a Goliath."
+
+/obj/effect/mob_spawn/corpse/watcher/crushed
+	corpse_description = "Crushed by a rockslide, it seemed to have been scraping frantically at the rocks even as it perished."
+	naive_corpse_description = "All of those rocks probably don't make a comfortable blanket."
+
+
+#define WATCHER_EGG_LIVELY_MOD 0.75
+#define WATCHER_EGG_ACTIVE_MOD 0.5
+
+/// Egg which hatches into a helpful pet. Or you can eat it if you want.
+/obj/item/reagent_containers/food/snacks/egg/watcher
+	name = "watcher egg"
+	desc = "A lonely egg still pulsing with life, somehow untouched by the corruption of the Necropolis."
+	icon_state = "egg_watcher"
+	tastes = list("ocular fluid" = 6, "loneliness" = 1)
+	antable = FALSE
+	/// How far have we moved?
+	var/steps_travelled = 0
+	/// How far should we travel to hatch?
+	var/steps_to_hatch = 600
+	/// Datum used to measure our steps
+	var/datum/movement_detector/pedometer
+
+/obj/item/reagent_containers/food/snacks/egg/watcher/Initialize(mapload)
+	. = ..()
+	pedometer = new(src, CALLBACK(src, PROC_REF(on_stepped)))
+
+/obj/item/reagent_containers/food/snacks/egg/watcher/Destroy(force)
+	. = ..()
+	QDEL_NULL(pedometer)
+
+/obj/item/reagent_containers/food/snacks/egg/watcher/examine(mob/user)
+	. = ..()
+	if(steps_travelled < (steps_to_hatch * WATCHER_EGG_ACTIVE_MOD))
+		. += "<span class='notice'>Something stirs listlessly inside.</span>"
+	else if(steps_travelled < steps_to_hatch * WATCHER_EGG_LIVELY_MOD)
+		. += "<span class='notice'>Something is moving actively inside.</span>"
+	else
+		. += "<span class='boldnotice'>It's jiggling wildly, it's about to hatch!</span>"
+
+
+
+/// Called when we are moved, whether inside an inventory or by ourself somehow
+/obj/item/reagent_containers/food/snacks/egg/watcher/proc/on_stepped(atom/movable/egg, atom/mover, atom/old_loc, direction)
+	var/new_loc = get_turf(egg)
+	if(isnull(new_loc) || new_loc == get_turf(old_loc))
+		return // Didn't actually go anywhere
+	steps_travelled++
+	if(steps_travelled == steps_to_hatch * WATCHER_EGG_ACTIVE_MOD)
+		jiggle()
+		return
```

Comment:
I dont really understand when this is supposed to occur, could you add a comment explaining?

Reply:
sure, it just starts it jiggling half way, added to code